test/libsolidity/syntaxTests/using/library_functions_attached_in_single_directive_inside_library.sol

```diff
@@ -0,0 +1,15 @@
+library L {
+    function publicFunction(uint) public pure {}
+    function externalFunction(uint) external pure {}
+    function internalFunction(uint) internal pure {}
+    using {publicFunction, externalFunction, internalFunction} for uint;
+    function f() public pure {
```

Comment:
Not sure if this function should be part of the test. The external and public functions are not supposed to work here, right? Or is this a bug?

Reply:
Why not? Public functions are fine and resolve to internal calls inside the library.

By the way, you should also add a private function here because these do work in libraries.